internal/services/subscription/data_source_locations.go

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package subscription
+
+import (
+
+	// nolint: staticcheck
+
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	resourcesSubscription "github.com/hashicorp/go-azure-sdk/resource-manager/resources/2022-12-01/subscriptions"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/subscription/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+var _ sdk.DataSource = ManagerSubscriptionLocationsDataSource{}
+
+type ManagerSubscriptionLocationsDataSource struct{}
+
+type ManagerSubscriptionLocationsDataSourceModel struct {
+	Location     string                             `tfschema:"location"`
+	RegionType   string                             `tfschema:"region_type"`
+	DisplayName  string                             `tfschema:"display_name"`
+	ZoneIds      []string                           `tfschema:"zone_ids"`
+	ZoneNames    []string                           `tfschema:"zone_names"`
+	ZoneMappings []SubscriptionLocationsZoneMapping `tfschema:"zone_mappings"`
+}
+
+type SubscriptionLocationsZoneMapping struct {
+	LogicalZone  string `tfschema:"logical_zone"`
+	PhysicalZone string `tfschema:"physical_zone"`
+}
+
+func (r ManagerSubscriptionLocationsDataSource) ResourceType() string {
+	return "azurerm_locations"
```

Comment:
maybe it should be named `azurerm_location` as it only gets one specific location data by `location`? And all `locations` should be replaced with `location`?

Reply:
fine for me